Add SPI Interface for BME680

jraber opened the pull request:
Addresses issues #10 (SPI not implemented)

I copied the SPI interface from the BME280 library and only slightly modified it.

I haven't tested this, as I don't have a BME680.  It would be great is someone with access to a BME680 could give it a try.

jraber committed:
Implement SPI interface

Mostly copied from the BME280 python library.

jraber committed:
Fix-up i2c_device import to make pylint happy

ladyada commented:
@brentru wanna try this? I know you had a BME680 recently!

brentru commented:
@ladyada  yep! I'll test it out today on hardware

brentru commented on adafruit_bme680.py:
```diff
@@ -324,7 +324,7 @@ class Adafruit_BME680_I2C(Adafruit_BME680):
     def __init__(self, i2c, address=0x77, debug=False, *, refresh_rate=10):
         """Initialize the I2C device at the 'address' given"""
-        import adafruit_bus_device.i2c_device as i2c_device
+        from adafruit_bus_device import i2c_device
```
Why are you changing the I2C import here?

ladyada replied:
its kinda the style we use in other drivers - so im not oppsoed to this change :D

jraber replied:
Hi @brentru. I changed it mostly to make pylint happy.

When I added the SPI code, pylint complained about the SPI import, so I "fixed" it.  Not wanting the library to have two different import styles, the choice was to either change the i2c import to match the new SPI import, which would make pylint happy, or to change the SPI import to match the i2c import. Also, I feel like the imports are a bit cleaner this way.

I put it in it's own commit to make it obvious that it has nothing to do with the SPI code.

brentru commented:
Output doesn't look correct on hardware. 

code.py output:
```
Temperature: -99.5137 degrees C
Gas: 12665144 ohms
Humidity: 0%
Pressure: -91.021hPa

```
Test Setup:
> Adafruit CircuitPython 4.0.0-beta.3 on 2019-03-07; Adafruit Feather M4 Express with samd51j19

jraber committed:
Improve SPI class, mirror the i2c class init and debug

Added the debug init parameter and copied the debug functionality
  from the i2c class
Added the refresh_rate initi parameter
Improved the SPI class docstring

jraber committed:
Change SPI memory page as needed when reading/writing through SPI

In SPI mode, we must change the memory page depending on which register is
  being read/written.

Renamed const _BME680_REG_STATUS to _BME680_REG_MEAS_STATUS to more closely
  match the name in the datasheet.

jraber commented:
@brentru Thanks for testing it.  It seems that the SPI interface on the BME680 is a little bit different from the BME280.  On the BME680, we have to toggle between two memory pages, depending on the register we are accessing.

Could you retest with the latest commit?  I added the memory page toggling.  If it still gives bad values, try with debug=True.  The debug data might help figure out what exactly is going on there.

makermelissa commented on adafruit_bme680.py:
```diff
@@ -349,3 +349,56 @@ def _write(self, register, values):
+        spi_mem_page = 0x00
+        if register < 0x80:
+            spi_mem_page = 0x10
+        self._write(_BME680_REG_STATUS, spi_mem_page)
```
Change this line to
`self._write(_BME680_REG_STATUS, bytes([spi_mem_page]))`

jraber replied:
I fixed it up.  Thanks testing and for the feedback.

makermelissa reviewed (changes requested):
Your change seems to fix the numbers, but the write function is expecting an array.

jraber committed:
values parameter for _write() must be a list

makermelissa commented:
Ok, it looks like it's working and the values were similar to the BME280.
